common/base/include/claragenomics/utils/device_buffer.hpp

```diff
@@ -187,14 +198,14 @@ class buffer
         using std::swap;
         swap(a.data_, b.data_);
         swap(a.size_, b.size_);
-        swap(a.stream_, b.stream_);
+        swap(a.streams_, b.streams_);
         swap(a.allocator_, b.allocator_);
     }
 
 private:
     value_type* data_;
     size_type size_;
-    cudaStream_t stream_;
+    std::vector<cudaStream_t> streams_;
```

Comment:
It is a bit redundant to have the associated streams here and in the allocator.
From what I understand this is only needed because of the `clear_and_resize()` and the `get_streams` methods, correct?
Wouldn't it be clearer if the `clear_and_resize()` method took the (potentially new) associated streams explicitly?

The only problematic case would be if `new_size == size_`. There you would need to add all new stream arguments to the memory block. Probably not better than the current solution, wanted to discuss it real quick nevertheless.

Reply:
I would not go for explicitly passing streams again. It could be that resizing is done in a subfunction, in that case the user would have to pass the stream explicitly

Reply:
Ok, let's leave it as it is then.